File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int n = heights.size();
        int ans = 0;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                int w = j - i;
                int l = min(heights[i],heights[j]);
                int area = l * w;
                ans = max(ans , area);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/coin-change/submission-0.java
class Solution {
    public int solve(int [] coins , int amount){
        if(amount == 0){
            return 0;
        }
        if(amount < 0){
            return Integer.MAX_VALUE;
        }

        int mini = Integer.MAX_VALUE;
        for(int coin : coins){
            int recans = solve(coins , amount-coin);
            if(recans != Integer.MAX_VALUE){
                int total = recans + 1;
                mini = Math.min(mini , total);
            }
        }
        return mini;
    }
    public int coinChange(int[] coins, int amount) {
        int ans = solve(coins , amount);
        if(ans == Integer.MAX_VALUE){
            return -1;
        }
        return ans;
    }     
}
